cpp/src/sampling/random_walks.cuh

```diff
@@ -1076,5 +1114,47 @@ std::
   return to_coo(d_v_view, d_sz_view);
 }
 
+/**
+ * @brief returns additional RW information on vertex paths offsets and weight path sizes and
+ * offsets, for the coalesced case (the padded case does not need or provide this information)
+ *
+ * @tparam index_t Type used to store indexing and sizes.
+ * @param handle RAFT handle object to encapsulate resources (e.g. CUDA stream, communicator, and
+ * handles to various CUDA libraries) to run graph algorithms.
+ * @param num_paths number of paths.
+ * @param ptr_d_sizes sizes of vertex paths.
+ * @return tuple of (vertex_path_offsets, weight_path_sizes, weight_path_offsets), where offsets are
+ * exclusive scan of corresponding sizes.
+ */
+template <typename index_t>
+std::tuple<rmm::device_uvector<index_t>, rmm::device_uvector<index_t>, rmm::device_uvector<index_t>>
+query_rw_sizes_offsets(raft::handle_t const& handle, index_t num_paths, index_t const* ptr_d_sizes)
+{
+  rmm::device_uvector<index_t> d_vertex_offsets(num_paths, handle.get_stream());
+  rmm::device_uvector<index_t> d_weight_sizes(num_paths, handle.get_stream());
+  rmm::device_uvector<index_t> d_weight_offsets(num_paths, handle.get_stream());
+
+  thrust::exclusive_scan(rmm::exec_policy(handle.get_stream())->on(handle.get_stream()),
+                         ptr_d_sizes,
+                         ptr_d_sizes + num_paths,
+                         d_vertex_offsets.begin());
+
+  thrust::transform(rmm::exec_policy(handle.get_stream())->on(handle.get_stream()),
+                    ptr_d_sizes,
+                    ptr_d_sizes + num_paths,
+                    d_weight_sizes.begin(),
+                    [] __device__(auto vertex_path_sz) { return vertex_path_sz - 1; });
+
+  CUDA_TRY(cudaStreamSynchronize(handle.get_stream()));
+
+  thrust::exclusive_scan(rmm::exec_policy(handle.get_stream())->on(handle.get_stream()),
```

Comment:
Just FYI, similarly, we'll eventually transition to ```rmm::exec_policy(handle.get_stream_view())``` instead of lengthier ```rmm::exec_policy(handle.get_stream())->on(handle.get_stream())```. Can be a 0.21 task.

Reply:
I see its use in one file only:  `cpp/tests/utilities/thrust_wrapper.cu`. If / when larger scope adoption happened in our code I would change it.